pkg/background/update_request_controller.go

```diff
@@ -90,7 +89,11 @@ func NewController(
 		UpdateFunc: c.updateUR,
 		DeleteFunc: c.deleteUR,
 	})
-	policyInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+	cpolInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: c.updatePolicy, // We only handle updates to policy
+		// Deletion of policy will be handled by cleanup controller
+	})
+	polInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
```

Comment:
I understand that we want to do the same for namespaced policies, but what was the reason we register the event handler for update events?

Correct me if I'm wrong - in the policy controller `pkg/policy`, we update/create UR on all types of policy events (CREATE/UPDATE/DELETE), and that triggers the reconciliation of the UR. Do we need this handler in the UR controller?

JFYI - there was an issue logged to decouple the above logic from the policy controller https://github.com/kyverno/kyverno/issues/3880.

Reply:
Sure! Adding @anushkamittal20 to this loop as she will be working on the issue.

Reply:
Or we move the issue to 1.8 ;)

Reply:
We'd like to get this fix in asap, 1.8 will be released in another 3 months:(

I can help with the cherry-pick!

Reply:
Let's get this in 1.7, I will take care of the cherry pick, I need to practice :laughing:

Reply:
1.7.0 or 1.7.1 ?